roles/addons/nic_nmcli/tasks/main.yml

```diff
@@ -65,7 +65,10 @@
     # {% endif %}
     # Allows to provide a simple ip4, or an ip4/prefix
     # Also support old prefix format (to be replaced by prefix4)
-    gw4: "{{ item.gw4 | default(networks[item.network].gateway4) | default(networks[item.network].gateway) | default(omit) }}"
+    gw4: "{{ (item.never_default4 | default(networks[item.network].never_default4) is not defined) | ternary( item.gw4 | default(networks[item.network].gateway4) | default(networks[item.network].gateway) | default(omit), omit) }}"
```

Comment:
Could you add a comment that explain this test with never_default4? Or explain this in the commit message?
On first read, this is not obvious.

Reply:
Added comment in the task.